app/views/session.py

```diff
@@ -35,15 +36,15 @@ def login():
     if session:
         session.clear()
 
-    logger.debug("attempting token authentication")
-    authenticator.jwt_login(request)
-    logger.debug("token authenticated - linking to session")
+    token = decrypt_token(request.args.get('token'))
+    if not use_token(token.get('jti')):
```

Comment:
Continuing previous discussion here. The RFC doesn't specify a format for the 'jti' just that it is unique; we (ONS) have defined the actual contents of it to be a GUID so should validate that what we received is correct as anything else suggests the sender hasn't implemented the spec correctly which is either bad implementation (could be broken; may fail or cause incorrect results ~ false positives etc.) or an indication of a compromised system. As a minimum we should check the length of the jti matches that of a a guid as defined in the schema def.